### ISSUE TITLE
Scanning for Quads Hangs

### ISSUE THREAD

**Issue by Boobies** [Cubitect/cubiomes-viewer#19]
Cubiomes Viewer hangs while searching for quads as of 1.3.1. However, 1.3.0 (released the same day so it's probably to do with the bug fix you did) does not hang. Using the Linux builds.

**Comment by Cubitect** [comment]
I don't see how the changes could have affected the quad-structure scanner and I can't reproduce this. What is the seed are you looking at?

**Comment by Boobies** [comment]
What I got was:

```
QMetaMethod::invoke: Dead lock detected in BlockingQueuedConnection: Receiver is MainWindow(0x7fff6e938898)
Killed
```
Perhaps it's due to a different version of the Qt library you linked against and it's not responding not because you're making it unresponsive but because there's a bug in the lib? That said, after failing pretty much consistently, It eventually worked on me and I'm not sure why cause nothing changed.

**Comment by Cubitect** [comment]
Okay I was able to reproduce the problem. This error message occurs when the event loop locks up, which means there was a problem in the thread logic. I had moved the proto-base computation (which is done before the search properly starts for the first time) into the main thread which caused a bunch of issues. I didn't clean up the proto-base lists for my tests so the issue didn't arise for me.

Hopefully the new release v1.3.2 should fixed the problem.
